tests/nn/pretrained_model_initializer_test.py

```diff
@@ -79,6 +81,23 @@ def test_from_params(self):
         assert initializer.weights
         assert initializer.parameter_name_overrides == name_overrides
 
+    def test_from_params_tar_gz(self):
+        try:
+            # Create temp.tar.gz
+            temp_tar_gz = os.path.join(self.TEST_DIR, "temp.tar.gz")
+            with tarfile.open(temp_tar_gz, "w:gz") as archive:
+                archive.add(self.temp_file, arcname=os.path.basename(self.temp_file))
+            params = Params({"type": "pretrained", "weights_file_path": temp_tar_gz})
+            initializer = Initializer.from_params(params)
+        finally:
+            os.remove(temp_tar_gz)
```

Comment:
Why not `with tempfile.NamedTemporaryFile(...):`?

This is an excellent place for a context manager :-)